src/p_spec.c

```diff
@@ -234,14 +241,7 @@ void P_HitFloor (mobj_t *mo)
     }
   }
 
-  if (terrain == terrain_water)
-    S_StartSound(mo, sfx_splash);
-  else if (terrain == terrain_slime)
-    S_StartSound(mo, sfx_ploosh);
-  else if (terrain == terrain_lava)
-    S_StartSound(mo, sfx_lvsiz);
-  else
-    S_StartSound(mo, sfx_oof);
+  S_StartSound(mo, hitsound[terrain, big_splash]);
```

Comment:
I didn't even know it works, see: https://en.cppreference.com/w/c/language/operator_other#Comma_operator
I think we should write `hitsound[terrain][big_splash]`

Reply:
Absolutely! That's what I did in a later fix.